tests/00-no-sleep.sh

```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+source "./helpers.bash"
+
+if grep -r "sleep" | grep -v "00-no-sleep"; then
+    echo "Please do not use sleep, consider using one of the wait helper functions."
+    echo "If none of the provided wait functions fit your use case please discuss your use case on Slack and / or file a bug."
+    exit 1
```

Comment:
I like this a lot.

I think we can keep this but ignore the helper.sh script and have a function `micro_sleep` which does `sleep 0.5` and we can insert this in all the loops which should relax the CPU a bit.